src/main/java/br/com/monkey/ecx/specification/SpecificationImpl.java

```diff
@@ -35,7 +36,7 @@ public Predicate toPredicate(Root<T> root, CriteriaQuery<?> criteriaQuery,
 		Path nestedRoot = getNestedRoot(root, Arrays.asList(nestedKey));
 		String criteriaKey = nestedKey[nestedKey.length - 1];
 
-		if (nestedRoot.get(criteriaKey).getJavaType().isEnum()) {
+		if (!isList(nestedRoot) && nestedRoot.get(criteriaKey).getJavaType().isEnum()) {
```

Comment:
essa validação aqui não tem necessidade.

Reply:
@FelipeAdorno aqui tenho um problema de quando a propriedade for List ocorre o erro ==> (Illegal attempt to dereference path source [null.minhaPropriedadeListNaClasse] of basic type)  ao tentar pegar a propriedade com ==> nestedRoot.get(criteriaKey), por isso coloquei a verificação para não entrar.

Tentei outras formas de verificação e a única que consegui foi verificando o packname se o mesmo contem o mesmo packname do root, ou seja, a propriedade que está sendo verificada
pertence a objetos da entidade raiz.

O if ficou da seguinte forma ==>
if (nestedRoot.getJavaType().getPackageName()
				.contains(root.getJavaType().getPackageName())
				&& nestedRoot.get(criteriaKey).getJavaType().isEnum()) {

Teria uma outra forma melhor para tratar isso?

Reply:
não entendi, ele entra nesse if mesmo verificando se é um enum?

Reply:
Li aqui de novo e entendi.